core/util/proxy.py

```diff
@@ -0,0 +1,108 @@
+"""
+OWASP Maryam!
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+#import requests
+from bs4 import BeautifulSoup
+import time
+#Python proxy module. Contributed by: @anuran-roy under @saeeddhqan
+class main:
+
+    def __init__(self):
+        self.framework.use = self.framework._global_options_['autoproxy']
+        self.framework.iplist = []
+
+    def getproxy(self):
+        #if self.use:
+        self.framework.output('[PROXY] Gathering proxies. It is quite time consuming, so do some pushups in the meantime :P')
+        purls = [ 'https://premproxy.com/list/' ]
+        i = 2
+        while i <= 8:
+            purl = f"https://premproxy.com/list/0{str(i)}.htm"
```

Comment:
Further analysis on premproxy.com reveals that you can sort the proxies by the time they were last accessed like [this](https://premproxy.com/list/time-01.htm).  
This would greatly increase your script's hit rate

Reply:
Sir I considered this, but then we won't get the anonymous proxies first 😅 

That brings me to my next question - Should we prioritize anonymity or availability for proxies? @saeeddhqan ?